codalab/worker/bundle_manager.py

```diff
@@ -299,23 +299,20 @@ def _deduct_worker_resources(self, workers_list):
         for worker in workers_list:
             for uuid in worker['run_uuids']:
                 bundle = self._model.get_bundle(uuid)
-                worker['cpus'] -= self._compute_request_cpus(bundle) or 1
-                worker['gpus'] -= self._compute_request_gpus(bundle) or 0
-                #TODO: memory_bytes
+                worker['cpus'] -= self._compute_request_cpus(bundle)
+                worker['gpus'] -= self._compute_request_gpus(bundle)
 
     def _filter_and_sort_workers(self, workers_list, bundle):
         """
         Filters the workers to those that can run the given bundle and returns
         the list sorted in order of preference for running the bundle.
-
-        Note: the worker slots field is no longer in use so it is effectively ignored
         """
 
         # keep track of which workers have GPUs
         has_gpu = {}
         for worker in workers_list:
             worker_id = worker['worker_id']
-            has_gpu[worker_id] = True if worker['gpus'] > 0 else False
+            has_gpu[worker_id] = worker['gpus'] > 0
```

Comment:
Still don't understand why `has_gpu` is needed...can't you always just write `worker['gpus'] > 0` instead?

Reply:
because later on self._deduct_worker_resources(workers_list) will reduce a particular worker's worker['gpus'] to 0 even though that worker does have a gpu (just not free)